ow_plexil/src/plexil-adapter/LanderInterface.cpp

```diff
@@ -50,56 +50,273 @@ static vector<string> LanderOpNames = {
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/camera_faults_status", QueueSize,
+                &LanderInterface::cameraFaultCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/arm_joint_accelerations", QueueSize,
+                &LanderInterface::armJointAccelCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/arm_joint_positions", QueueSize,
+                &LanderInterface::armJointPositionCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/arm_joint_torques", QueueSize,
+                &LanderInterface::armJointTorqueCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/arm_joint_velocities", QueueSize,
+                &LanderInterface::armJointVelocityCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/arm_pose", QueueSize,
+                &LanderInterface::armPoseCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/pan_tilt_position", QueueSize,
+                &LanderInterface::panTiltCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/battery_state_of_charge", QueueSize,
+                &LanderInterface::batteryChargeCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/battery_temperature", QueueSize,
+                &LanderInterface::batteryTempCb, this)));
+
+  m_subscribers.push_back
+    (make_unique<ros::Subscriber>
+     (m_genericNodeHandle ->
+      subscribe("/battery_remaining_useful_life", QueueSize,
+                &LanderInterface::batteryLifeCb, this)));
+}
+
+///////////////////////// Subscriber Callbacks ///////////////////////////////
+
+void LanderInterface::batteryChargeCb
+(const owl_msgs::BatteryStateOfCharge::ConstPtr& msg)
+{
+  m_battery_soc = msg->value;
+  publish ("BatteryStateOfCharge", m_battery_soc);
+}
+
+void LanderInterface::batteryLifeCb
+(const owl_msgs::BatteryRemainingUsefulLife::ConstPtr& msg)
+{
+  // NOTE: This is not being called as of 4/12/21.  Jira OW-656 addresses.
+  m_battery_rul = msg->value;
+  publish ("BatteryRemainingUsefulLife", m_battery_rul);
+}
+
+void LanderInterface::batteryTempCb
+(const owl_msgs::BatteryTemperature::ConstPtr& msg)
+{
+  m_battery_temp = msg->value;
+  publish ("BatteryTemperature", m_battery_temp);
+}
+
+void LanderInterface::armJointAccelCb
+(const owl_msgs::ArmJointAccelerations::ConstPtr& msg)
+{
+  copy(msg->value.begin(), msg->value.end(), m_arm_joint_accelerations.begin());
```

Comment:
Good catch.  I have added loops that publish the arm telemetry for each index, since the Lookups read values for just one joint.  I think the publish rate is slow enough for this not to pose a performance hit.